### PR TITLE
Hazelcast native support fixes #1647

### LINKED ISSUE #1647: Hazelcast native support
There is a Hazelcast Client extension by Hazelcast https://github.com/hazelcast/quarkus-hazelcast-client

### PR DESCRIPTION
**Pull request by zbendhiba**
[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

**Comment by zbendhiba** [comment]
Please note that this PR doesn't include all the integration tests needed. I'll follow up with an other PR: see issue https://github.com/apache/camel-quarkus/issues/2094

**Comment by ppalaga** [comment]
I have restarted the CI

File: integration-tests/hazelcast/src/main/java/org/apache/camel/quarkus/component/hazelcast/it/HazelcastRoutes.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    @Produces
+    @ApplicationScoped
+    @Unremovable
+    @Named("hazelcast-topic")
+    HazelcastDefaultComponent hazelcastTopic() {
+        final HazelcastTopicComponent hazelcastComponent = new HazelcastTopicComponent();
+        return getHazelcastComponent(hazelcastComponent);
+    }
+
+    private HazelcastDefaultComponent getHazelcastComponent(HazelcastDefaultComponent hazelcastComponent) {
```
Comment (ppalaga) [inline integration-tests/hazelcast/src/main/java/org/apache/camel/quarkus/component/hazelcast/it/HazelcastRoutes.java:129]: `configureHazelcastComponent` would perhaps better describe what this method does.

**Review by ppalaga — APPROVED** [review]
What an amount of work! Thanks! Two minor nitpicks inline that you may or may not want to address in the follow up.

File: extensions/hazelcast/runtime/pom.xml [hunk 2]
```diff
@@ -55,6 +63,20 @@
             <groupId>org.apache.camel</groupId>
             <artifactId>camel-hazelcast</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.hazelcast</groupId>
+            <artifactId>quarkus-hazelcast-client</artifactId>
+        </dependency>
+        <!-- fix version -->
```
[the remark this answers is not in the document]
Comment (zbendhiba) [reply extensions/hazelcast/runtime/pom.xml:70]: nope!

**Pull request merged**